packages/common/src/GenericContract.ts

```diff
@@ -0,0 +1,616 @@
+
+    constructor(
+        protected grpcClient: ConcordiumGRPCClient,
+        protected contractAddress: ContractAddress,
+        protected contractName: string,
+        protected schema?: Schema<E>
+    ) {
+        this.dryRunInstance = new ContractDryRun(
+            grpcClient,
+            contractAddress,
+            contractName
+        );
+    }
+
+    /**
+     * Helper function for getting the {@link InstanceInfo} of a contract
+     *
+     * @param {ConcordiumGRPCClient} grpcClient - The GRPC client for accessing a node.
+     * @param {ContractAddress} contractAddress - The address of the contract.
+     *
+     * @throws if the {@link InstanceInfo} of the contract could not be found.
+     *
+     * @returns {InstanceInfo} the instance info.
+     */
+    protected static async getInstanceInfo(
+        grpcClient: ConcordiumGRPCClient,
+        contractAddress: ContractAddress
+    ): Promise<InstanceInfo> {
+        const instanceInfo = await grpcClient.getInstanceInfo(contractAddress);
+
+        if (instanceInfo === undefined) {
+            throw new Error(
+                `Could not get contract instance info for contract at address ${stringify(
+                    contractAddress
+                )}`
+            );
+        }
+
+        return instanceInfo;
+    }
+
+    /**
+     * Helper function for getting the name of a contract
+     *
+     * @param {ConcordiumGRPCClient} grpcClient - The GRPC client for accessing a node.
+     * @param {ContractAddress} contractAddress - The address of the contract.
+     *
+     * @throws if the {@link InstanceInfo} of the contract could not be found.
+     *
+     * @returns {string} the name of the contract.
+     */
+    protected static async getContractName(
+        grpcClient: ConcordiumGRPCClient,
+        contractAddress: ContractAddress
+    ): Promise<string> {
+        const instanceInfo = await this.getInstanceInfo(
+            grpcClient,
+            contractAddress
+        );
+        return getContractName(instanceInfo);
+    }
+
+    /**
+     * A dry-run instance, providing access to methods for performing dry-run invocations of update instructions.
+     */
+    public get dryRun(): ContractDryRun<E> {
+        return this.dryRunInstance;
+    }
+
+    /**
+     * Creates a {@link ContractUpdateTransactionWithSchema} contract update transaction, holding the necessary parts to sign/submit to the chain.
+     *
+     * @param {string} entrypoint - The name of the receive function to invoke.
+     * @param {Function} serializeInput - A function to serialize the `input` to bytes.
+     * @param {ContractTransactionMetadata} metadata - Metadata to be used for the transaction creation (with defaults).
+     * @param {T} input - Input for for contract function.
+     *
+     * @throws If the query could not be invoked successfully.
+     *
+     * @returns {ContractUpdateTransaction} The transaction hash of the update transaction
```

Comment:
Is this correct? it might throw if the parameter is incorrect, but it does not send the transaction